src/app/ui/dock.cpp

```diff
@@ -422,13 +540,101 @@ bool Dock::onProcessMessage(ui::Message* msg)
     case kMouseUpMessage: {
       if (hasCapture()) {
         releaseMouse();
-        onUserResizedDock();
+        const auto* mouseMessage = static_cast<MouseMessage*>(msg);
+
+        if (m_dropzonePlaceholder && m_dropzonePlaceholder->parent()) {
+          // Always undock the dropzone placeholder to avoid dangling sizes.
+          auto* placeholderCurrentDock = dynamic_cast<Dock*>(m_dropzonePlaceholder->parent());
+          placeholderCurrentDock->undock(m_dropzonePlaceholder.get());
+        }
+
+        if (m_hit.dockable) {
+          auto* dockableWidget = dynamic_cast<Widget*>(m_hit.dockable);
+          auto* widgetDock = dynamic_cast<Dock*>(dockableWidget->parent());
+
+          int currentSide = widgetDock->whichSideChildIsDocked(dockableWidget);
+
+          assert(dockableWidget && widgetDock);
+
+          auto dockNRoll = [&](const int side) {
+            const gfx::Rect workspaceBounds = widgetDock->bounds();
+
+            gfx::Size size;
+            if (dockableWidget->id() == "timeline") {
+              size.w = 64;
+              size.h = 64;
+              auto timelineSplitterPos = get_config_double(kLegacyLayoutMainWindowSection,
+                                                           kLegacyLayoutTimelineSplitter,
+                                                           75.0) /
+                                         100.0;
+              auto pos = gen::TimelinePosition::LEFT;
+              size.w = (workspaceBounds.w * (1.0 - timelineSplitterPos)) / guiscale();
+
+              if (side & RIGHT) {
+                pos = gen::TimelinePosition::RIGHT;
+              }
+              if (side & BOTTOM) {
+                pos = gen::TimelinePosition::BOTTOM;
+                size.h = (workspaceBounds.h * (1.0 - timelineSplitterPos)) / guiscale();
+              }
```

Comment:
If side is TOP, we can save the old timeline configuration as bottom, just for backward compatibility.